scheduler/reconciler/reconcile_node.go

```diff
@@ -183,6 +215,27 @@ func (nr *NodeReconciler) diffSystemAllocsForNode(
 			continue
 		}
 
+		// find nodes on which to deploy canaries
+		canaryNode := slices.ContainsFunc(
+			canaryNodes[tg.Name],
+			func(n *structs.Node) bool { return n.ID == alloc.NodeID },
+		)
```

Comment:
Minor item: given this is the only consumer of `canaryNodes`, maybe it would make more sense to return `map[string]map[string]*structs.Node` (where `map[tgName] -> map[nodeID] -> node`) so we have a set and don't have to loop over the potentially large nodes list repeatedly?

Reply:
[27d166b](https://github.com/hashicorp/nomad/pull/26499/commits/27d166bd457725238c9693c73fb3c0d16586b9a0)